app/src/main/java/com/example/calculator/myapplication/ControlUnit.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.myapplication
+import com.example.myapplication.Command.Command
+
+class ControlUnit {
+    private val commands = mutableListOf<Command>()
+    private var current: Int = 0
+    fun StoreCommand(command: Command){
+        commands.add(command)
+    }
+    fun ExecuteCommand(){
+        commands[current].Execute()
+        current++
+    }
+    fun Undo(levels: Int){
+        for (i in 0 until levels){
+            if(current > 0) commands[--current].UnExecute()
+        }
```

Comment:
можно без if, так будет проще 
```suggestion
        val availableLevels = minOf(current, levels)
        for (i in 0 until availableLevels) {
            commands[--current].UnExecute()
        }
```